modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerAbstractConnectionContext.java

```diff
@@ -104,33 +95,12 @@ public GridKernalContext kernalContext() {
     /**
      * Perform authentication.
      *
-     * @return Auth context.
      * @throws IgniteCheckedException If failed.
      */
-    protected AuthorizationContext authenticate(GridNioSession ses, String user, String pwd)
-        throws IgniteCheckedException {
-        if (ctx.security().enabled())
-            authCtx = authenticateExternal(ses, user, pwd).authorizationContext();
-        else if (ctx.authentication().enabled()) {
-            if (F.isEmpty(user))
-                throw new IgniteAccessControlException("Unauthenticated sessions are prohibited.");
-
-            authCtx = ctx.authentication().authenticate(user, pwd);
+    protected void authenticate(GridNioSession ses, String user, String pwd) throws IgniteCheckedException {
```

Comment:
There is a weird behavior on last line of this method. We do a check of secCtx for NULL, if NULL that throws an exception. But actually AuthProcessor never returns NULL, it just throws exception on auth failure. Also docs for GridSecurityProcessor are misleading:
```

     * @return {@code True} if succeeded, {@code false} otherwise.
    public SecurityContext authenticate(AuthenticationContext ctx) throws IgniteCheckedException;
```
Let's fix declaration of return value in the docs (true / false -> SecurityContext), also there should be a description for auth fail behavior - should it return NULL or throw an exception.

Reply:
We should not consider AuthProcessor as the only security implementation. It can be replaced with the custom implementation by the user and  nothing restricts null to be returned. So the check is justified.
I have fixed java doc.